NEWS.md

```diff
@@ -478,6 +478,16 @@ Deprecated or removed
   * The no-op `transpose` fallback has been deprecated. Consider introducing suitable
     `transpose` methods or calling `permutedims(x, [2,1])` ([#13171], [#17075], [#17374]).
 
+  * The following macros have been deprecated ([#18189]):
+    * `@windows` is deprecated in favor of `@static is_windows()`
+    * `@unix` is deprecated in favor of `@static is_unix()`
+    * `@osx` is deprecated in favor of `@static is_apple()`
+    * `@linux` is deprecated in favor of `@static is_linux()`
+    * `@windows_only` is deprecated in favor of `@static if is_windows()`
+    * `@unix_only` is deprecated in favor of `@static if is_unix()`
+    * `@osx_only` is deprecated in favor of `@static if is_apple()`
+    * `@linux_only` is deprecated in favor of `@static if is_linux()`
```

Comment:
These shouldn't mention `@static`, Mentioning that using `@static` could be useful/necessary when used in a function's local scope at the end is ok.